parachain/primitives/core/src/outbound.rs

```diff
@@ -129,18 +141,68 @@ impl Command {
 		}
 	}
 
+	/// Compute gas cost
+	/// reference gas from benchmark report with some extra margin for incentive
+	/// | Function Name    | min    | avg    | median | max    | # calls |
+	/// | agentExecute    | 487    | 5320   | 3361   | 14074  | 4       |
+	/// | createAgent    | 839    | 184709 | 237187 | 237187 | 9       |     
+	/// | createChannel    | 399    | 31023  | 2829   | 75402  | 5       |
+	/// | updateChannel    | 817    | 15121  | 3552   | 36762  | 5     |     
+	/// | transferNativeFromAgent    | 770    | 21730  | 21730  | 42691  | 2       |
+	/// | setOperatingMode    | 682    | 12838  | 13240  |  24190  | 4       |
+	/// | upgrade    | 443    | 9270   | 3816   | 29004  | 4       |
+	pub fn dispatch_gas(&self) -> u128 {
+		match self {
+			Command::CreateAgent { .. } => 300000,
+			Command::CreateChannel { .. } => 100000,
+			Command::UpdateChannel { .. } => 50000,
+			Command::TransferNativeFromAgent { .. } => 60000,
+			Command::SetOperatingMode { .. } => 40000,
+			Command::AgentExecute { command, .. } => match command {
+				AgentExecuteCommand::TransferToken { .. } => 30000,
+			},
+			// leave enough space for upgrade with arbitrary initialize logic
+			Command::Upgrade { .. } => 300000,
+		}
+	}
+
+	pub fn base_fee_required(&self) -> bool {
+		match self {
+			Command::CreateAgent { .. } => true,
+			Command::CreateChannel { .. } => true,
+			Command::AgentExecute { .. } => true,
```

Comment:
I agree that `AgentExecute` should require a base fee. But the notion document says otherwise. See [comment](https://www.notion.so/Fees-c15d9159574c4ff1b1f82bdf220d7f94?d=8371b178e0c84f0c96d3ed96658375fe&pvs=4#2e741cda15f747238678134a345cbdf2).

Reply:
The meaning of "base fee" in this context is that the sender needs to pay a small static fee for the weight involved in processing outbound messages locally.

The "extra fee" is the cost of dispatching on Ethereum, but as you can see from the Notion doc we don't always want to charge this fee on BridgeHub.

Reply:
For `AgentExecute` the reward is paid from the agent which is maintained by parachain themselves.